File: WarehouseManager/src/main/java/hu/elte/WarehouseManager/model/Product.java
package hu.elte.WarehouseManager.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Data;

import javax.persistence.*;
import java.util.List;

@Entity
@Data
@Table(name = "PRODUCTS")
public class Product {

    @Id
    @Column(name = "PRODUCT_ID")
    private Integer productID;

    @Column(name = "PRODUCT_NAME")
    private String productName;

    @Column(name = "PRODUCT_TYPE")
    private String productType;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "PRODUCT_GROUP_ID")
    private ProductGroup productGroup;

    @Column(name = "QUANTITY_IN_STOCK")
    private Integer quantity;

    @OneToMany(fetch = FetchType.EAGER, mappedBy = "product")
    @JsonIgnore
    private List<Request> requests;
}

File: WarehouseManager/warehouse-manager-client/src/app/model/product.ts
import { ProductGroup } from '../model/productgroup';

export class Product {

    // fields
    productID: number;
    productName: string;
    productType: string;
    productGroup: ProductGroup;
    quantity: number;

    // constructor
    constructor(
        productID:number,
        productName:string,
        productType:string,
        productGroup:ProductGroup,
        quantity:number) { 

            this.productID = productID; 
            this.productName = productName;
            this.productType = productType;
            this.productGroup = productGroup;
            this.quantity = quantity;
    }
}